addons/web/static/src/views/form/form_controller.js

```diff
@@ -288,15 +290,32 @@ export class FormController extends Component {
      * is invalid or if a server error is thrown.
      * @param {Record} record
      */
-    async onRecordSaved(record) {}
+    async onRecordSaved(record) {
+        if (this.duplicateId === record.id && this.translationChanges) {
+            this.duplicateId = undefined;
+            await this.orm.call(this.model.root.resModel, "override_translations", [
+                [this.model.root.resId],
+                this.translationChanges,
+            ]);
+        }
+    }
 
     /**
      * onWillSaveRecord is a callBack that will be executed before the
      * record save if the record is valid if the record is valid.
      * If it returns false, it will prevent the save.
      * @param {Record} record
      */
-    async onWillSaveRecord(record) {}
+    async onWillSaveRecord(record) {
+        if (this.duplicateId === record.id) {
+            this.translationChanges = {};
+            for (const fieldName in record._changes) {
+                if (record.fields[fieldName].translate) {
+                    this.translationChanges[fieldName] = record._changes[fieldName];
```

Comment:
This value may have problem for `update_field_translations` when `callable(field.translate)`(python server) which requires `translation` to be the format `{lang: {old_term: new_term}}`

But in the frontend, I think you cannot know that from `record.fields[fieldName].translate`

Maybe do the similar thing like this POC of mine
https://github.com/odoo/odoo/pull/113514
check the diff for api `update_field_translations`
```python
def update_field_translations(self, fname, translations, reset_langs=None):
```
and remember to change the corresponding js code for the translation dialog to make sure you can still invalidate translation by fill an empty value from the translation dialog.